From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Tom Urlings <tom.urlings(at)gmail(dot)com>
Date: Tue May 27 12:45:35 PDT 2025
Subject: Enable monitor command setexpr w/fmt and sub/gsub/regex options

This will enable bootscripts to calculate load address dynamically during boot,
removing the need for updating load addresses whenever kernel/initial ramdisk 
change size.

Also, this will enable bootscripts to find device tree files in folder structures
that contain a vendor path component, e.g. dtb/allwinnner/...
---
 configs/nanopi-r2s-rk3328_defconfig | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/configs/nanopi-r2s-rk3328_defconfig b/configs/nanopi-r2s-rk3328_defconfig
index cafb38ffcee..7941af4ff2e 100644
--- a/configs/nanopi-r2s-rk3328_defconfig
+++ b/configs/nanopi-r2s-rk3328_defconfig
@@ -34,11 +34,12 @@ CONFIG_SPL_ATF=y
 CONFIG_SPL_ATF_NO_PLATFORM_PARAM=y
 CONFIG_CMD_BOOTZ=y
 CONFIG_CMD_GPT=y
 CONFIG_CMD_MMC=y
 CONFIG_CMD_USB=y
-# CONFIG_CMD_SETEXPR is not set
+CONFIG_CMD_SETEXPR=y
+CONFIG_CMD_SETEXPR_FMT=y
 CONFIG_CMD_TIME=y
 CONFIG_SPL_OF_CONTROL=y
 CONFIG_TPL_OF_CONTROL=y
 CONFIG_OF_SPL_REMOVE_PROPS="clock-names interrupt-parent assigned-clocks assigned-clock-rates assigned-clock-parents"
 CONFIG_TPL_OF_PLATDATA=y
@@ -94,8 +95,9 @@ CONFIG_USB_OHCI_GENERIC=y
 CONFIG_USB_DWC2=y
 CONFIG_USB_DWC3=y
 # CONFIG_USB_DWC3_GADGET is not set
 CONFIG_USB_GADGET=y
 CONFIG_USB_GADGET_DWC2_OTG=y
+CONFIG_REGEX=y
 CONFIG_SPL_TINY_MEMSET=y
 CONFIG_TPL_TINY_MEMSET=y
 CONFIG_ERRNO_STR=y
-- 
Created with Armbian build tools https://github.com/armbian/build

